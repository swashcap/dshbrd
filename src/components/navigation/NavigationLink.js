// @flow
import * as React from "react";
import { NavLink } from "react-router-dom";
import styled from "styled-components";

import VisuallyHidden from "../VisuallyHidden";

const ACTIVE_CLASS_NAME = "active";

type Props = {
  children?: React.Node,
  to: string
};

/**
 * Make react-router's `NavLink` play well with styled-components.
 */
const _NavigationLink = ({ children, to, ...rest }: Props) => (
  <NavLink activeClassName={ACTIVE_CLASS_NAME} exact to={to} {...rest}>
    <VisuallyHidden>{children}</VisuallyHidden>
  </NavLink>
);

const NavigationLink = styled(_NavigationLink)`
  cursor: pointer;
  display: block;
  padding: 1em 0.667em;

  &:focus::before,
  &:hover::before {
    background: rgba(0, 0, 0, 0.1);
  }
  &:active::before {
    background: rgba(0, 0, 0, 0.2);
  }

  &::before {
    background: transparent;
    border-color: rgba(0, 0, 0, 0.5);
    border-radius: 100%;
    border-style: solid;
    border-width: 1px;
    content: "";
    display: inline-block;
    height: 0.5em;
    line-height: 1;
    transition-duration: 200ms;
    duration-property: background border-color;
    vertical-align: middle;
    width: 0.5em;
  }

  &.${ACTIVE_CLASS_NAME}:focus::before, &.${ACTIVE_CLASS_NAME}:hover::before {
    background: black;
  }
  &.${ACTIVE_CLASS_NAME}:active::before {
    background: black;
  }
  &.${ACTIVE_CLASS_NAME}::before {
    background: black;
    border-color: black;
  }
`;

export default NavigationLink;
