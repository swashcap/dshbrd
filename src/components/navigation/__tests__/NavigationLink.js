// @flow
import * as React from "react";
import TestRenderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import NavigationLink from "../NavigationLink";

test("renders correctly", () => {
  const testRenderer = TestRenderer.create(
    <MemoryRouter>
      <NavigationLink to="/">Home</NavigationLink>
    </MemoryRouter>
  );
  expect(testRenderer).toMatchSnapshot();
});
