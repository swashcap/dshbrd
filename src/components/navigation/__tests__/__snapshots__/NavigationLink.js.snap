// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
  white-space: nowrap;
}

.c0 {
  cursor: pointer;
  display: block;
  padding: 1em 0.667em;
}

.c0:focus::before,
.c0:hover::before {
  background: rgba(0,0,0,0.1);
}

.c0:active::before {
  background: rgba(0,0,0,0.2);
}

.c0::before {
  background: transparent;
  border-color: rgba(0,0,0,0.5);
  border-radius: 100%;
  border-style: solid;
  border-width: 1px;
  content: "";
  display: inline-block;
  height: 0.5em;
  line-height: 1;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  duration-property: background border-color;
  vertical-align: middle;
  width: 0.5em;
}

.c0.active:focus::before,
.c0.active:hover::before {
  background: black;
}

.c0.active:active::before {
  background: black;
}

.c0.active::before {
  background: black;
  border-color: black;
}

<a
  aria-current="page"
  className="c0 active"
  href="/"
  onClick={[Function]}
  style={Object {}}
>
  <span
    className="c1"
  >
    Home
  </span>
</a>
`;
